cloudsync/tasks.py

```diff
@@ -95,14 +132,36 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
 
     # Generate thumbnails for the 1st encoding (no point in doing so for each).
     outputs[0]['ThumbnailPattern'] = THUMBNAIL_PATTERN.format(video_file.s3_basename)
-    transcoder = Transcoder(
+    transcoder = TechTVTranscoder(
         settings.ET_PIPELINE_ID,
         settings.AWS_REGION,
         settings.AWS_ACCESS_KEY_ID,
         settings.AWS_SECRET_ACCESS_KEY
     )
-    transcoder.encode(video_input, outputs, Playlists=playlists)
-    transcoder.create_job_for_object(video)
+    try:
+        transcoder.encode(video_input, outputs, Playlists=playlists)
+    except ClientError as e:
+        log.exception('Transcode job creation failed for video %s: %s', video_id, e.response)
+        update_video_status(video, VideoStatus.TRANSCODE_FAILED)
+        self.update_state(task_id=task_id, state=states.FAILURE)
+        transcoder.message = e.response
+        raise e
+    finally:
+        job = transcoder.create_job_for_object(video)
+
+    # Wait for the transcoding job to complete
+    waiter = transcoder.client.get_waiter('job_complete')
+    try:
+        waiter.wait(Id=transcoder.message['Job']['Id'])
```

Comment:
It makes `read_job` requests to AWS every 30 seconds to see if the transcode job has finished.  So it keeps the celery task running until then.  The more I think about it the less I like this approach.  I would prefer to have a separate periodic celery task that would run every X seconds/minutes to check on the job status instead.